File: test/export/vb/07-subprocesos.vb
Module SIN_TITULO
	Sub Main()
		Dim c As Double
		Console.WriteLine(funciondoble(5))
		noretornanada(3,9)
		c = 0
		porreferencia(c)
		Console.WriteLine(c)
		Dim a(10) As String
		Dim b(3,4) As String
		recibevector(a)
		recibematriz(b)
	End Sub
	Public Function funciondoble(ByVal b As Double)
		Dim a As Double
		a = 2*b
		Return a
	End Function
	Public Sub noretornanada(ByVal a As Double, ByVal b As Double)
		Console.WriteLine(a+b)
	End Sub
	Public Sub porreferencia(ByRef b As Double)
		b = 7
	End Sub
	Public Sub recibevector(ByVal v() As String)
		v(1) = Console.ReadLine()
	End Sub
	Public Sub recibematriz(ByVal m(,) As String)
		m(1,1) = Console.ReadLine()
	End Sub
End Module

File: test/export/cs/07-subprocesos.cs
using System;
namespace PSeInt {
	class sin_titulo {
		static void Main(string[] args) {
			double c;
			Console.WriteLine(funciondoble(5));
			noretornanada(3,9);
			c = 0;
			porreferencia(ref c);
			Console.WriteLine(c);
			string[] a = new string[10];
			string[,] b = new string[3,4];
			recibevector(a);
			recibematriz(b);
		}
		static double funciondoble(double b) {
			double a;
			a = 2*b;
			return a;
		}
		static void noretornanada(string a, string b) {
			Console.WriteLine(a+b);
		}
		static void porreferencia(ref double b) {
			b = 7;
		}
		static void recibevector(string[] v) {
			v[0] = Console.ReadLine();
		}
		static void recibematriz(string[,] m) {
			m[0,0] = Console.ReadLine();
		}
	}
}
